pkg/apis/operations/v1alpha1/imageprepull_types.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2023 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePrepullJob is used to prepull images on edge node.
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type ImagePrepullJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the specification of the desired behavior of ImagePrepullJob.
+	// +required
+	Spec ImagePrepullJobSpec `json:"spec"`
+
+	// Status represents the status of ImagePrepullJob.
+	// +optional
+	Status ImagePrepullJobStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePrepullJobList is a list of ImagePrepullJob.
+type ImagePrepullJobList struct {
+	// Standard type metadata.
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard list metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// List of ImagePrepullJob.
+	Items []ImagePrepullJob `json:"items"`
+}
+
+// ImagePrepullSpec represents the specification of the desired behavior of ImagePrepullJob.
+type ImagePrepullJobSpec struct {
+	// ImagePrepullTemplate represents original templates of imagePrepull
+	ImagePrepullTemplate ImagePrepullTemplate `json:"imagePrepullTemplate,omitempty"`
```

Comment:
There is no other place reference to this template, so no need to define it?

Reply:
> There is no other place reference to this template, so no need to define it?

Templates are defined for adding the overrider capability in later versions.